core/src/main/java/com/klaytn/caver/tx/type/AbstractTxType.java

```diff
@@ -84,14 +85,59 @@ public AbstractTxType(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit
         this.from = from;
         this.to = to;
         this.value = value;
+        this.senderSignatureDataList = new ArrayList();
     }
 
     protected void setSenderSignatureData(KlaySignatureData signatureData) {
-        this.senderSignatureData = signatureData;
+        addSenderSignatureData(signatureData);
     }
 
     public KlaySignatureData getSenderSignatureData() {
-        return senderSignatureData;
+        try {
+            return senderSignatureDataList.get(0);
+        } catch (Exception e){
+            return null;
+        }
+    }
+
+    protected void addSenderSignatureData(KlaySignatureData signatureData) {
+        for (KlaySignatureData senderSignatureData : senderSignatureDataList) {
+            if (senderSignatureData.equals(signatureData)) {
+                return;
+            }
+        }
+        senderSignatureDataList.add(signatureData);
+    }
+
+    public void addSenderSignatureDataList(List<KlaySignatureData> signatureDataList) {
+        for (KlaySignatureData signatureData : signatureDataList) {
+            addSenderSignatureData(signatureData);
+        }
+    }
+
+    protected void addSenderSignatureData(List<RlpType> signatureRlpTypeList) {
+        for (RlpType signature : signatureRlpTypeList) {
+            RlpList vrs = (RlpList) signature;
+            byte[] v = ((RlpString) vrs.getValues().get(0)).getBytes();
+            byte[] r = ((RlpString) vrs.getValues().get(1)).getBytes();
+            byte[] s = ((RlpString) vrs.getValues().get(2)).getBytes();
+            addSenderSignatureData(new KlaySignatureData(v, r, s));
+        }
+    }
+
+    public void addSignatureData(List<RlpType> values, int offset) {
+        if (values.size() > offset) {
+            List<RlpType> senderSignatures = ((RlpList) (values.get(offset))).getValues();
```

Comment:
same here. Don't we need to have `try` here?

Reply:
Doesn't need `try` because it checks and imports? What do you think?

Reply:
Doesn't the code need `try` because it checks the size of the values? What do you think?

Reply:
I just said because of `(values.get(offset))`. It's up to you.

Reply:
In `if (values.size ()> offset)`, the offset is guaranteed to be less than the size of the values, so it is not necessary to insert a try statement. Thank you.